pywincffi/kernel32/process.py

```diff
@@ -240,3 +240,34 @@ def TerminateProcess(hProcess, uExitCode):
         ffi.cast("UINT", uExitCode)
     )
     error_check("TerminateProcess", code=code, expected=Enums.NON_ZERO)
+
+
+def CreateToolhelp32Snapshot(dwFlags, th32ProcessID):
+    """
+    Takes a snapshot of the specified processes, as well as the heaps,
+    modules, and threads used by these processes.
+
+    .. seealso::
+
+        https://msdn.microsoft.com/en-us/ms682489
+
+    :param int dwFlags:
+        The portions of the system to be included in the snapshot.
+
+    :param int th32ProcessID:
+        The process identifier of the process to be included in the snapshot.
+
+    :return:
+        If the function succeeds,
+        it returns an open handle to the specified snapshot.
+    """
+    input_check("dwFlags", dwFlags, integer_types)
+    input_check("th32ProcessID", th32ProcessID, integer_types)
+    ffi, library = dist.load()
+    process_list = library.CreateToolhelp32Snapshot(
```

Comment:
You need to check to ensure `process_list` is not `INVALID_HANDLE_VALUE`.  If it is, you need to raise a WindowsAPIError.  It would look something like this:

```python
    if handle == INVALID_HANDLE_VALUE:  # pragma: no cover
        raise WindowsAPIError(
            "GetStdHandle", "Invalid Handle", INVALID_HANDLE_VALUE,
            expected_return_code="not %r" % INVALID_HANDLE_VALUE)
```

ref: https://github.com/opalmer/pywincffi/blob/1eb07d873f0afa1ffb238f24260b78e42ec9d166/pywincffi/kernel32/handle.py#L67